lib/failover_databases.rb

```diff
@@ -0,0 +1,61 @@
+class FailoverDatabases
+  FAILOVER_DATABASES_YAML_FILE = Rails.root.join("config", "failover_databases.yaml")
+
+  def self.refresh_databases_list
+    query_replication_manager
+  end
+
+  def self.all_databases
+    if File.exist?(FAILOVER_DATABASES_YAML_FILE)
+      begin
+        YAML.load(File.read(FAILOVER_DATABASES_YAML_FILE))
+      rescue => err
+        _log.error("#{err.class}: #{err}")
+        _log.error(err.backtrace.join("\n"))
+        []
+      end
+    else
+      query_replication_manager
+    end
+  end
+
+  def self.standby_databases
+    result = []
+    all_databases.each do |record|
+      result << record if record["type"] == 'standby'
+    end
+    result
+  end
+
+  def self.standby_and_active_databases
+    result = []
+    all_databases.each do |record|
+      result << record if record["type"] == 'standby' && record["active"] == true
+    end
+    result
+  end
+
+  def self.query_replication_manager
```

Comment:
This might actually be more clear if it was called `query_repmgr`. They always have it abbreviated in their docs and everything and the schema is very specific to the particular tool we are using.